src/providers/splash.ts

```diff
@@ -0,0 +1,53 @@
+import { AmountWithDecimal, Provider } from "@/types/core";
+import { AmmPool, makeNativePools, Pools } from "@ergolabs/ergo-dex-sdk";
+import { AssetAmount, Explorer, RustModule } from "@ergolabs/ergo-sdk";
+
+class Splash implements Provider {
+  private baseUrl: string;
+
+  constructor(url: string) {
+    this.baseUrl = url;
+  }
+
+  async getOrderBook(marketId: string) {
+    try {
+      const response = await fetch(
+        `${this.baseUrl}/platform-api/v1/trading-view/order-book?base=${marketId}&quote=.`,
+        {
+          method: "GET",
+          headers: {
+            "Content-Type": "application/json",
+          },
+        }
+      );
+      return response.json();
+    } catch (error) {
+      console.error("Error fetching order book:", error);
+      throw new Error("Failed to fetch order book");
+    }
+  }
+
+  async x2y(marketId: string, amount: AmountWithDecimal) {
+    const orderBook = await this.getOrderBook(marketId);
+
+    const ask = orderBook.asks[0].price;
+
+    return {
+      amount: ask.toFixed(amount.decimals),
```

Comment:
There are two important issues here:
1. When we say token X has amount of `1000` with decimal of `2`, it means the human readable format will be `10.00`, which is simply `10`.
2. `amount.decimals` is the decimals for input value. The returned value should have its own decimals, based on the token. For example, ADA has a decimal of 6, but ERG a decimal of 9. It seems that Splash uses the following endpoint for getting token data:
https://spectrum.fi/cardano-token-list-v2.json